plugins/module_utils/network/sonic/config/bgp_communities/bgp_communities.py

```diff
@@ -342,27 +373,120 @@ def get_new_add_request(self, conf):
-                    if 'members' not in conf:
-                        conf['members'] = item['members']
+            if cur_state == "merged":
+                for item in have:
+                    if item['name'] == conf['name']:
+                        if 'type' not in conf:
+                            conf['type'] = item['type']
+                        if 'permit' not in conf or conf['permit'] is None:
+                            conf['permit'] = item['permit']
+                        if 'match' not in conf:
+                            conf['match'] = item['match']
+                        if conf['type'] == "standard":
+                            for attr in self.standard_communities_map:
+                                if attr not in conf and attr in item:
+                                    conf[attr] = item[attr]
+                        else:
+                            if 'members' not in conf:
+                                if item.get('members', {}) and item['members'].get('regex', []):
+                                    conf['members'] = {'regex': item['members']['regex']}
+                                else:
+                                    conf['members'] = item['members']
+                        break
+
             new_req = self.get_new_add_request(conf)
             if new_req:
                 requests.append(new_req)
         return requests
+
+    def get_replaced_overridden_config(self, want, have, cur_state):
+        commands, requests = [], []
+
+        commands_del, requests_del = [], []
+        commands_add, requests_add = [], []
+
+        for conf in want:
+            name = conf['name']
+            in_have = False
+            for have_conf in have:
+                if have_conf['name'] == name:
+                    in_have = True
+                    if have_conf['type'] != conf['type']:
+                        # If both community list are of same name but different types
+                        commands_del.append(have_conf)
+                        commands_add.append(conf)
+                    else:
+                        is_change = is_delete = False
+
+                        if have_conf['permit'] != conf['permit']:
+                            is_change = is_delete = True
+
+                        if have_conf['match'] != conf['match']:
+                            is_change = is_delete = True
+
+                        if conf["type"] == "standard":
+                            no_attr = True
+                            for attr in self.standard_communities_map:
+                                if not conf.get(attr, None):
+                                    if have_conf.get(attr, None):
+                                        is_delete = True
+                                else:
+                                    no_attr = False
+                                    if not have_conf.get(attr, None):
+                                        is_change = True
```

Comment:
Please handle the scenario where options in 'want' are a subset of options in 'have'.

Reply:
Updated with a example.